content/2017.03.building-projects-as-a-way-to-learn.rst

```diff
@@ -0,0 +1,258 @@
+them too much power. In the meantime these companies are offering a
+service that solve people problems, and they get addicted to it and
+come back for more.
+
+We alone cannot compete with companies that are spending millions of
+dollars building great products to capture users.
+
+However, we can build software that, on the one hand, is useful for
+the Mozilla mission and enable us to fix Mozilla issues quickly while
+building Firefox and, on the other hand, empowers developers to build
+great software where users choose where they want to store and control
+their data.
+
+What matters rather than building software is the vision that this
+software carries.
+
+If you don't have a vision, what are you building? Sure, writing lines
+of code is cheap and fast. You can build a prototype in two weeks and
+iterate until you are happy with the Minimum Viable Product.
+
+Note that, it might not be so cheap anymore when you start adding
+documentation and maintenance costs.
+
+However, the product might be used, and might get some traction for
+months or even years but it will be eventually forgotten in a git
+repository somewhere forever.
+
+Writing software to solve problems is not enough, to keep motivated on
+the long term. You need a greater goal than just writing code.
+
+Knowing that, how do you keep motivated building things that are
+thrown away?
+
+Different people or communities might have different guiding questions.
+
+For us the question is: As well as solving the problem, what can we
+learn while building this solution that can be useful for us and for
+our community to help us go faster and build more robust software
+later.
+
+For Mozilla the question is: How do I improve Firefox
+so that people want to use it as their main browser?
+
+Both questions lead to goals that we set to ourselves. We are using
+software development to reach them.
+
+Mozilla is calling that the True North, we are calling it a vision,
+some other people might call it a utopia.
+
+In both cases that vision should set the route of our ambition and
+point out a direction for everyone to sail to.
+
+How do you set course for a vision?
+===================================
+
+.. image:: {filename}/images/draken.jpg
+    :alt: Draken sailing to the North
+    :align: center
+
+
+|OKRs| are a way to cut your vision in steps and to move your vision
+forward.
+
+To be able to define OKRs from your vision, you need to be able to find
+the fastest way to get closer to your aim, however what we can learn
+from sailing is that the fastest way is rarely the shortest.
+
+At Mozilla, we have a vision that is written down in the
+`Mozilla Manifesto <https://www.mozilla.org/en-US/about/manifesto/>`_.
+
+Mitchell Baker does a great job making sure everyone at Mozilla stays
+aligned with that vision.
+
+However agreeing on the fastest way and turning the Mozilla vision
+into OKRs is really challenging and even more when you have a thousand
+people's opinion about it.
```

Comment:
Both sentences doesn't carry the same idea. In one we are talking about people in the other one we are talking about ideas, one people can have multiple ideas but what I want to underline is that each people as a specific idea of what the manifesto should be.